framework/app.mk

```diff
@@ -257,11 +256,18 @@ else
   app_plugin_deps :=
 endif
 
-app_LIBS       := $(app_LIB) $(app_LIBS)
-app_LIBS_other := $(filter-out $(app_LIB),$(app_LIBS))
+# For a variable that has been previously defined, += honors the type that
+# variable has been defined as, i.e., either recursively-expanded (=) or
+# simply-expanded (:=). For variables that have not been previously defined, +=
+# will behave as =. In those cases, we instead prepend explicitly by using :=.
```

Comment:
tbh I was thinking more 'app_LIBS' is used for this, we must keep track of that

than explaining how `:=` and `+=` work which I unfortunately know having had to fight Makefiles before

Reply:
Yeah, this was more to explain why we use `+=` in some circumstances and not use it in others, which might not be completely obvious looking at the list of `app_` variables. :) Should I still delete it?

Reply:
yup pls